ale/kernel_access.py

```diff
@@ -295,3 +298,65 @@ def remove_dups(listofElements):
     else:
         warnings.warn(f"{format_as} is not a valid format, returning as dict")
         return kernels
+
+
+def spiceql_call(function_name = "", function_args = {}, use_web=False):
+    """
+    Interface to SpiceQL (Spice Query Library) for both Offline and Online use
+
+    This function will access the value passed through props defined as `web`. This
+    value determines the access pattern for spice data. When set to Online, you will
+    access the SpiceQL service provided through the USGS Astro AWS platform. This service
+    performs kernel and data aquisition. If set to Offline, you will access locally loaded
+    kernels, and SpiceQL will do no searching for you.
+
+    Parameters
+    ----------
+    functions_name : str
+                        String defineing the function to call, properly exposed SpiceQL
+                        functions should map 1-to-1 with endpoints on the service
+                    
+    function_args : dict
+                    Dictionary of arguments used by the function
+
+    Returns : any
+                Any return from a SpiceQL function
+    """
+    if not use_web:
+        func = getattr(pyspiceql, function_name)
+
+        # Ensure that in offline mode we anticipate the user loading/passing their own kernels
+        # to ALE
+        return func(**function_args)
+
+    url = "https://spiceql-dev.prod-asc.chs.usgs.gov/v1/"
+    url += function_name
+    headers = {
+        'accept': '*/*',
+        'Content-Type': 'application/json'
+    }
+
+    r = requests.get(url, params=function_args, headers=headers, verify=False)
+    r.raise_for_status()
+    response = r.json()
+
+    if r.status_code != 200:
```

Comment:
Kinda sorta, this is a weird quirk of our lambda setup that I think needs to change. Basically if the lambda function called fails, we still get a "valid" web return as we catch the failure.

I think we can just let the function itself fail and that should be handled as expected(?)

It may also be that all of our endpoints only handle returning 200 codes.

This likely needs to be ironed out before a full release